src/GafferScene/Constraint.cpp

```diff
@@ -217,7 +230,15 @@ boost::optional<ScenePlug::ScenePath> Constraint::targetPath() const
 	ScenePath targetPath;
 	ScenePlug::stringToPath( targetPathAsString, targetPath );
 
-	if( !inPlug()->exists( targetPath ) )
+	const ScenePlug *targetScene = targetScenePlug();
+	if( !targetScene->getInput() )
+	{
+		// Backwards compatibility for time when there was
+		// no `targetScene` plug.
```

Comment:
Very minor thing, but is it backwards compatibility, or just desired functionality? Feels like it would be a shame to remove it at some point and force the double connection for standard operations? I think we do this with `CopyAttributes` though don't we?

Reply:
I wasn't sure to be honest, but I was leaning towards it being backwards compatibility. If designing from scratch, I think I would have matched `CopyAttributes` and always used the second input unconditionally. I prefer things with simple explanations where possible, and it's easier to explain that the `targetScene` is always the scene where the `target` is found. I don't see the double connection as being too troublesome for simple setups, but I don't feel too strongly either way. I doubt we'll ever get to the point of removing this compatibility though.